src/handlers/user-identity/tests/user-identity-handler.test.ts

```diff
@@ -157,3 +163,114 @@ describe("user-identity-handler tests", () => {
+      fraudValidityPeriod: TEST_FRAUD_VALIDITY_HOURS,
+    } as Configuration);
+    jest.spyOn(fraudCheckService, "hasFraudCheckExpired").mockRestore();
+
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date(NOW));
+  });
+
+  it.each([
+    { fraudCheckInputs: [{ nbf: NOT_EXPIRED_NBF, state: State.CURRENT }], expectedExpired: false },
+    { fraudCheckInputs: [{ nbf: EXPIRED_NBF, state: State.CURRENT }], expectedExpired: true },
+    {
+      fraudCheckInputs: [
+        { nbf: NOT_EXPIRED_NBF, state: State.CURRENT },
+        { nbf: EXPIRED_NBF, state: State.HISTORIC },
+      ],
+      expectedExpired: false,
+    },
+    {
+      fraudCheckInputs: [
+        { nbf: EXPIRED_NBF, state: State.CURRENT },
+        { nbf: NOT_EXPIRED_NBF, state: State.HISTORIC },
+      ],
+      expectedExpired: true,
+    },
+  ])(`should set expired value based on nbf of CURRENT fraud check`, async ({ fraudCheckInputs, expectedExpired }) => {
+    const fraudChecks = fraudCheckInputs.map((input) => {
+      return { vc: createIdentityCheckCredentialJWT(input.nbf, FRAUD_ISSUER), state: input.state };
+    });
+    const passportCheck = createIdentityCheckCredentialJWT(RANDOM_NBF, PASSPORT_ISSUER);
+
+    const mockCredentialStoreData = createCredentialStoreIdentityResponse([
+      ...fraudChecks,
+      { vc: passportCheck, state: State.CURRENT },
+    ]);
+    mockEVCSResponse(mockCredentialStoreData);
+
+    const result = await handler(newEvent, {} as Context);
+
+    expect(result.statusCode).toBe(HttpCodesEnum.OK);
+    const body = JSON.parse(result.body) as UserIdentityResponseMetadata;
+    expect(body.vot).toBe("P2");
+    expect(body.content).toEqual({
+      sub: "user-sub",
+      vot: "P2",
+      vtm: [],
+    });
+    expect(body.expired).toBe(expectedExpired);
+    expect(body.isValid).toBe(true);
+    expect(body.kidValid).toBe(true);
+    expect(body.signatureValid).toBe(true);
+  });
+});
+
+const createCredentialStoreIdentityResponse = (
+  verifiableCredentialStates: { vc: VerifiableCredentialJWT; state: State }[] = []
+): CredentialStoreIdentityResponse => {
+  const storedIdentity = {
+    sub: "user-sub",
+    vot: "P2",
+    vtm: [],
+  };
+
+  return {
+    si: {
+      state: State.CURRENT,
+      vc: sign(getDefaultStoredIdentityHeader(), storedIdentity),
+      metadata: null,
+    },
+    vcs: verifiableCredentialStates.map((vcState) => {
+      return { state: vcState.state, vc: sign(getDefaultStoredIdentityHeader(), vcState.vc), metadata: null };
+    }),
+  };
+};
+
+const createIdentityCheckCredentialJWT = (nbfDate: string, issuer: string): IdentityCheckCredentialJWTClass => {
+  const nbfSeconds = getDateSeconds(new Date(nbfDate));
+  return {
+    iss: issuer,
+    nbf: nbfSeconds,
+    sub: "sdf",
+    vc: {
+      evidence: [{}],
+      type: ["VerifiableCredential", "IdentityCheckCredential"],
+    },
+  };
+};
+
+const getDateSeconds = (date: Date): number => {
+  return date.getTime() / 1000;
```

Comment:
To avoid decimal places coming in:
```suggestion
  return Math.floor(date.getTime() / 1000);
```